src/Compiler/TypedTree/tainted.fs

```diff
@@ -101,11 +101,12 @@ type internal Tainted<'T> (context: TaintedContext, value: 'T) =
             | :? TypeProviderError -> reraise()
             | :? AggregateException as ae ->
                     let errNum,_ = FSComp.SR.etProviderError("", "")
-                    let messages = [for e in ae.InnerExceptions -> e.Message]
+                    let messages = [for e in ae.InnerExceptions -> if e.InnerException = null then e.Message else (e.Message + ": " + e.GetBaseException().Message)]
```

Comment:
Don't use generic equality for null-checks:
```suggestion
                    let messages = [for e in ae.InnerExceptions -> if isNull e.InnerException then e.Message else (e.Message + ": " + e.GetBaseException().Message)]
```

It does feel like a bit of a hack to only do this for one level of inner exceptions, though. For instance, if this is wrapped in an `AggregateException` you would still loose the benefit of this change.

Reply:
Actually, just checked [`GetBaseException()`](https://learn.microsoft.com/en-us/dotnet/api/system.exception.getbaseexception?view=net-8.0). It already does what you want. So, instead, this does the same:
```suggestion
                    let messages = [for e in ae.InnerExceptions -> if e.GetBaseException().Message)]
```

Reply:
I'd favor showing the full stack trace, sometimes the inner message is helpful, but it will not be as precise as the full stacktrace.

Maybe this can be guarded behind an environment variable (so we don't have to surface it as compiler flag)?